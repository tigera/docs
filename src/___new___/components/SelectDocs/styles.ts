export default {};

export const sectionOuterStyles = (hasDarkBg: boolean) => ({
  flexDirection: 'column',
  h: 'auto',
  w: '100%',
  bg: hasDarkBg ? 'tigeraBlack' : 'tigeraGrey.100',
  pb: '60px',
});

export const stackStyle = {
  justifyContent: 'center',
  h: 'auto',
  alignItems: 'center',
  mb: -2,
  px: [4, 0],
};

export const headerTextStyle = {
  textAlign: 'center',
  pt: 14,
  mb: 4,
};

export const subHeaderTextStyle = {
  fontWeight: '600',
  textAlign: 'center',
  mt: 7,
  fontSize: '2xl',
  lineHeight: 9,
};

export const innerTextStyle = {
  textAlign: 'center',
  fontSize: 'xl',
  fontWeight: 'normal',
  color: 'tigeraGrey.800',
  mx: '44px',
  mb: 10,
  mt: 2,
  lineHeight: 8,
};

export const iconContainerStyle = {
  display: 'flex',
  justifyContent: 'center',
  pt: '44px',
};

export const iconStyle = {
  width: 32,
  height: 32,
};

export const rectangleStyle = {
  minW: ['290px', '371px'],
  border: '1px',
  borderColor: 'tigeraCloudGrey',
  borderRadius: 'lg',
  bg: 'tigeraWhite',
  ml: [0, 4],
  mr: [0, 4],
  mt: [4, 4, 4, 4, ''],
  mb: [4, 4, 4, 4, ''],
  pb: '40px',
};

export const actionBoxStyles = {
  mb: 14,
  justifyContent: 'center',
};
