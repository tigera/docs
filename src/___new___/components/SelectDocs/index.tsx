import * as React from 'react';
import { Box, Flex, Text, SystemStyleObject, Stack, Heading, Image, Link } from '@chakra-ui/react';
import { heading2Styles } from '../styles';
import selectDocsInfo from '../../data/selectDocsInfo';

import {
  sectionOuterStyles,
  iconContainerStyle,
  iconStyle,
  rectangleStyle,
  stackStyle,
  subHeaderTextStyle,
  headerTextStyle,
} from './styles';

interface SelectDocsProps {
  sx?: SystemStyleObject;
  isDarkMode: boolean;
}

const SelectDocs: React.FC<SelectDocsProps> = ({ isDarkMode, ...rest }) => (
  <Flex
    sx={sectionOuterStyles(isDarkMode)}
    {...rest}
  >
    <Heading
      as='h2'
      size='md'
      sx={{ ...heading2Styles(isDarkMode), ...headerTextStyle }}
    >
      Select a doc set
    </Heading>
    <Stack
      sx={stackStyle}
      direction={['column', 'column', 'column', 'column', 'row', 'row', 'row']}
    >
      {selectDocsInfo.selectDocsInfo.map((info, index) => (
        <Box key={index}>
          <Link href={info.link}>
            <Box
              sx={rectangleStyle}
              key={index}
            >
              <Box sx={iconContainerStyle}>
                <Image
                  src={info.image}
                  sx={iconStyle}
                  loading='lazy'
                />
              </Box>
              <Text sx={subHeaderTextStyle}>{info.title}</Text>
            </Box>
          </Link>
        </Box>
      ))}
    </Stack>
  </Flex>
);

export default SelectDocs;
