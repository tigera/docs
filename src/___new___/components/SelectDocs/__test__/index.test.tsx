import { render } from '@testing-library/react';
import SelectDocs from '../index';

describe('<SelectDocs/>', () => {
  it('should render the HowItWorks component', () => {
    const { asFragment } = render(<SelectDocs isDarkMode={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
