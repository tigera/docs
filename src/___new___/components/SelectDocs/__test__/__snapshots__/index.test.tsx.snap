// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectDocs/> should render the HowItWorks component 1`] = `
<DocumentFragment>
  <div
    class="css-15xukzj"
  >
    <h2
      class="chakra-heading css-17pfrz0"
    >
      Select a doc set
    </h2>
    <div
      class="chakra-stack css-1mtxuht"
    >
      <div
        class="css-0"
      >
        <div
          class="css-yqjfh3"
        >
          <div
            class="css-1ki4hma"
          >
            <img
              class="chakra-image css-x3wokz"
              loading="lazy"
              src="img/calico-logo-badge.svg"
            />
          </div>
          <p
            class="chakra-text css-3o8wkr"
          >
            Calico Open Source
          </p>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-yqjfh3"
        >
          <div
            class="css-1ki4hma"
          >
            <img
              class="chakra-image css-x3wokz"
              loading="lazy"
              src="/img/calico-enterprise-logo-badge.svg"
            />
          </div>
          <p
            class="chakra-text css-3o8wkr"
          >
            Calico Enterprise
          </p>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-yqjfh3"
        >
          <div
            class="css-1ki4hma"
          >
            <img
              class="chakra-image css-x3wokz"
              loading="lazy"
              src="/img/calico-cloud-logo-badge.svg"
            />
          </div>
          <p
            class="chakra-text css-3o8wkr"
          >
            Calico Cloud
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
